### ISSUE TITLE
[question] is there an option/custom type for Date to handle number directly?

### ISSUE THREAD

**Issue by huafu** [Automattic/mongoose#7175]
**Do you want to request a *feature* or report a *bug*?**
That would be a feature if it doesn't exist yet

**What is the current behavior?**
Codewise (javascript side and not mongodb), `Date` fields are read/written as instances of `Date`, `Number` fields are read/written as `number`.

**What is the expected behavior?**
This is of course the expected behavior, but what if I want to have my `ISODate` of mongodb handled as timestamps in my code (ie `number`)? I know I can `+doc.dateField`, but it'd be nice if there was an option with the `Date` type so that it returns `number` directly.

If there is such an option already, I couldn't find it (I guess due to the 50 so-called timestamp plugins which are just about `createdAt` and `updatedAt` 🤣) – could be nice then to put it in the docs, or better index the docs if it's there as I missed it as well :-/

**Comment by vkarpov15** [comment]
I'm not sure I understand the question. Just to clarify, I think you're asking for [Mongoose's built-in timestamps](https://mongoosejs.com/docs/guide.html#timestamps) to support storing updatedAt/createdAt as a number instead of a date, so something like:

```javascript
new Schema({
  createdAt: Number // instead of `Date` like how mongoose timestamps currently do
});
```

Is that correct?

**Comment by huafu** [comment]
@vkarpov15 no, I am NOT talking about the `timestamps` option. I just want a `Date` field (which is `ISODate` in mongodb) to be used as a Number in js (ie timestamp).

But I actually got an answer on SO which matches the behavior I was looking for:

```js
new Schema({
  answeredAt: {
    type: Date,
    get: val => +val,
    set: val => new Date(val),
  },
});
```

Leaving this open as I'd like to know if there is an easier way to achieve this (apart from making it a special Type or so ;-)
